src/Microsoft.ApiDesignGuidelines.Analyzers/Core/OperatorOverloadsHaveNamedAlternates.Fixer.cs

```diff
@@ -26,11 +28,130 @@ public sealed override FixAllProvider GetFixAllProvider()
             return WellKnownFixAllProviders.BatchFixer;
         }
 
-        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
-        {                              
-            // This is to get rid of warning CS1998, please remove when implementing this analyzer
-            await new Task(() => { });
-            throw new NotImplementedException();
+        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            return Task.Run(() => context.RegisterCodeFix(new MyCodeAction(ct => Fix(context, ct)), context.Diagnostics.First()));
+        }
+
+        private static async Task<Document> Fix(CodeFixContext context, CancellationToken cancellationToken)
+        {
+            var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            var root = await context.Document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(context.Span);
+            var generator = context.Document.Project?.LanguageServices?.GetService<SyntaxGenerator>();
+            if (generator == null)
+            {
+                return context.Document;
+            }
+
+            var diagnostic = context.Diagnostics.First();
+            switch (diagnostic.Properties[OperatorOverloadsHaveNamedAlternatesAnalyzer.DiagnosticKindText])
+            {
+                case OperatorOverloadsHaveNamedAlternatesAnalyzer.AddAlternateText:
+                    var methodDeclaration = generator.GetDeclaration(node, DeclarationKind.Operator) ?? generator.GetDeclaration(node, DeclarationKind.ConversionOperator);
+                    var operatorOverloadSymbol = (IMethodSymbol)semanticModel.GetDeclaredSymbol(methodDeclaration, context.CancellationToken);
```

Comment:
does this work? the method declaration is what you just created, and symantic model is from original document. semantic model will not have the method decl node you just created.